### PR TITLE
Fix/sdl stops video when hmi doesnt respond audiostream

### PR DESCRIPTION
**Pull request by OlhaVorobiova**
Fixes #[12300](https://adc.luxoft.com/jira/browse/FORDTCN-12300)
This PR is **ready** for review.

### Risk
This PR makes **minor** API changes.

### Testing Plan
 [ATF test scripts](https://github.com/LuxoftSDL/sdl_atf_test_scripts/commit/cc84d5b9b9b959b17c0c4657f60a708a0fdd6aeb)

### Summary
I've added function EndService(uint32_t app_id, protocol_handler::ServiceType service_type) where sdl stops streaming for one servise (audio or video) and checks a state of another service. In case if both services stops, sdl  will unregister application.

### CLA
- [x] I have signed [the CLA](https://docs.google.com/forms/d/e/1FAIpQLSdsgJY33VByaX482zHzi-xUm49JNnmuJOyAM6uegPQ2LXYVfA/viewform)

### CODE REVIEW

**Comment by ychernysheva** [comment]
@OlhaVorobiova May I kindly suggest to request adding some brief summary of introduced changes? In description of this PR, it's a good practice, maybe instead of duplication of reproduction steps, which are also provided in test script

**Review by ychernysheva — APPROVED** [review]

File: src/components/application_manager/include/application_manager/application_manager_impl.h [hunk 1]
```diff
@@ -847,11 +847,19 @@ class ApplicationManagerImpl
...
    */
   void EndNaviServices(uint32_t app_id) OVERRIDE;
 
+  /**
+   * @brief Ends opened navi service audio or video for application
+   * @param app_id Application id
+   * @param service_type Service type to check
+   */
+  void EndService(uint32_t app_id,
+                  protocol_handler::ServiceType service_type) OVERRIDE;
```
Comment (AKalinich-Luxoft) [inline src/components/application_manager/include/application_manager/application_manager_impl.h:861]:
```suggestion
  void EndService(const uint32_t app_id,
                  const protocol_handler::ServiceType service_type) OVERRIDE;
```

File: src/components/application_manager/include/application_manager/application_manager_impl.h [hunk 2]
```diff
@@ -1439,7 +1447,8 @@ class ApplicationManagerImpl
    * opened services. Streaming ability could be restored by AllowStreaming();
    * @param app_id Application to proceed
    */
-  void DisallowStreaming(uint32_t app_id);
+  void DisallowStreaming(uint32_t app_id,
+                         protocol_handler::ServiceType service_type);
```
Comment (AKalinich-Luxoft) [inline src/components/application_manager/include/application_manager/application_manager_impl.h:1451]:
```suggestion
  void DisallowStreaming(const uint32_t app_id,
                         const protocol_handler::ServiceType service_type);
```

Comment (AKalinich-Luxoft) [reply src/components/application_manager/include/application_manager/application_manager_impl.h:1451]: Also please add new parameter to description

**Review by AKalinich-Luxoft — APPROVED** [review]

**Review by VadymLuchko — APPROVED** [review]

**Comment by OlhaVorobiova** [comment]
Reopened after comment in PR [3752](https://github.com/smartdevicelink/sdl_core/pull/3752#pullrequestreview-727943950)

**Pull request closed without merging**